lib/pool.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const storage = Symbol('storage');
+
+class Pool {
+  constructor(factory) {
+    this.factory = factory;
+    this[storage] = [];
+  }
+
+  put(value) {
+    this[storage].push(value);
+  }
+
+  get() {
```

Comment:
Perhaps we should allow passing arguments to `factory` here
i.e.
```js
get(...args) {
  ...
  return this.factory(...args);
}
```

Reply:
How would that work? The idea behind the `Pool` is that you always get the same objects, if you need to pass arguments to the factory, you can just bind the factory function before passing it to the constructor or setting the `factory` field of the `Pool` instance to it.
Besides, it would be confusing when you are calling `get()` with arguments, and expect to get an object constructed with those arguments, but instead you get an object that was created before with other arguments and put to the `Pool` instance.